packages/valory/skills/abstract_round_abci/behaviours.py

```diff
@@ -247,8 +249,15 @@ def teardown(self) -> None:
 
     def act(self) -> None:
         """Implement the behaviour."""
-        self._process_current_round()
+        if not self.tm_manager.is_tm_unhealthy or self.tm_manager.is_acting:
+            # tendermint is not healthy, or we are already applying a fix.
+            # try_fix() internally uses generators, that's why it's relevant
+            # to know whether a fix is already being applied.
+            # It might happen that tendermint is healthy, but the fix is not yet finished.
+            self.tm_manager.try_fix()
+            return
 
+        self._process_current_round()
```

Comment:
the check now is concurrently with the operation of the behaviour.

Reply:
This check ensures that the behaviours are run when tendermint is healthy. I couldn't think of a reason why we would want to have it otherwise.